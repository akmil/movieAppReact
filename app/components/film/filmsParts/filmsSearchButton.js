import React from 'react';
import {Button} from '../../common/button';
import PropTypes from 'prop-types'

const FilmsSearchButtonâ€¨ = props => {
    const stylesWhite = {
        background: 'white',
        border: 'none',
        color: '#F44336',
        padding: '10px'
    };
    const theme = (props.theme) ? `${props.theme}-theme` : '';

    return <div className={`${theme} col-66 text-align-right`}>
            <Button
                onClick = {props.onClick}
                style={stylesWhite}
            >{props.name}</Button>
        </div>
};

FilmsSearchButton.propTypes = {
    name: PropTypes.string.isRequired,
    theme: PropTypes.string.isRequired,
    // onClick: PropTypes.func.isRequired
};

export default FilmsSearchButton