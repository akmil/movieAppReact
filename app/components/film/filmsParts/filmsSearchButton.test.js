import React from 'react';
import { shallow } from 'enzyme';
import {FilmsSearchButton} from './filmsSearchButton';
// import renderer from 'react-test-renderer';

describe('<films FilmsSearchButton />', () => {
    it('should render SearchButton without errors', () => {
        const wrapper = shallow(<FilmsSearchButton/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should set class .dark-theme', () => {
        const wrapper = shallow(<FilmsSearchButton theme={'dark'}/>);
        console.log(wrapper.find('.dark-theme'));
        expect(wrapper.find('.dark-theme').length).toEqual(1);
    })
});