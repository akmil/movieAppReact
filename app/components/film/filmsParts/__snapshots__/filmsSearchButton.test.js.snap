// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<films FilmsSearchButton /> should render SearchButton without errors 1`] = `
<div
  className=" col-66 text-align-right"
>
  <Button
    style={
      Object {
        "background": "white",
        "border": "none",
        "color": "#F44336",
        "padding": "10px",
      }
    }
  />
</div>
`;
